### ISSUE TITLE
Ability to add local directory as source

### ISSUE THREAD

**Issue by neilharris123** [argoproj/argo-cd#3432]
# Summary

In the application resource YAML file, we are able to add a source repo and path.
It would be useful to be able to specify the local directory for testing locally.

# Motivation

I'm creating CI tests and automated dev deployment scripts which are triggered in the PR. The container I am using already contains a replica/clone of the git branch the test will run against. I would like to change certain values in manifests in the containers local directory. But when I run argocd cli commands against the application resource file in the local directory, because the source repoURL value must point at the remote git branch, the local changes are never propagated as I would like.

# Proposal

I was thinking something like the following would satisfy my needs:

```
spec:
  project: default
  source:
    dirPath: /path/to/dir
```
or 

```
spec:
  project: default
  source:
    repoURL: local
    targetRevision: HEAD
    path: guestbook

**Comment by alexmt** [comment]
Hello @neilharris123 ,

Some Argo CD CLI commands already supports using local directory:

compare live app state and local manifests source:
```
argocd app diff <appName> --local /path/to/dir
```

sync app using local manifests
```
argocd app sync <appName> --local /path/to/dir
```

Does it cover your use cases?

**Comment by neilharris123** [comment]
Hello @alexmt,

It does support my use case up to a point because in theory I can create the app initially from the repo, then sync to add the local changes. However, I've noticed when I perform a sync using the --local option that I get this error, because syncing from local, unlike from repo, does not seem to support kustomize build options. 

`: external plugins disabled; unable to load external plugin 'SimpleSOPS'`

It appears support for sync from repo this was added [here](https://github.com/argoproj/argo-cd/pull/1817), but does not work when using the --local parameter. Unless I'm missing something.
